### PR TITLE
Disable libxar

### PR DESCRIPTION
**Pull request by h-vetinari**
Plus a couple of clean-ups & a re-render.

To hopefully unblock https://github.com/conda-forge/lld-feedstock/pull/33.

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -13,18 +13,18 @@ cmake -G "Ninja" ^
     -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
     -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% ^
     -DLLVM_USE_INTEL_JITEVENTS=ON ^
+    -DLLVM_ENABLE_LIBXML2=ON ^
+    -DLLVM_ENABLE_RTTI=ON ^
+    -DLLVM_ENABLE_ZLIB=ON ^
+    -DLLVM_INCLUDE_BENCHMARKS=OFF ^
```
Comment (h-vetinari) [inline recipe/bld.bat:19]: Added.

File: recipe/bld.bat [hunk 2]
```diff
@@ -13,18 +13,18 @@ cmake -G "Ninja" ^
...
+    -DLLVM_INCLUDE_DOCS=OFF ^
     -DLLVM_INCLUDE_EXAMPLES=OFF ^
     -DLLVM_INCLUDE_TESTS=OFF ^
     -DLLVM_INCLUDE_UTILS=ON ^
     -DLLVM_INSTALL_UTILS=ON ^
     -DLLVM_UTILS_INSTALL_DIR=libexec\llvm ^
-    -DLLVM_INCLUDE_DOCS=OFF ^
-    -DLLVM_ENABLE_RTTI=ON ^
-    -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly ^
-    -DLLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN=ON ^
```
Comment (h-vetinari) [inline recipe/bld.bat:24]: Removed.

File: recipe/build.sh [hunk 3]
```diff
@@ -24,21 +24,22 @@ fi
 
 cmake -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
       -DCMAKE_BUILD_TYPE=Release \
+      -DHAVE_LIBEDIT=OFF \
+      -DLLVM_ENABLE_LIBXML2=OFF \
       -DLLVM_ENABLE_RTTI=ON \
-      -DLLVM_INCLUDE_TESTS=ON \
+      -DLLVM_ENABLE_TERMINFO=OFF \
+      -DLLVM_ENABLE_ZLIB=ON \
+      -DLLVM_INCLUDE_BENCHMARKS=OFF \
```
Comment (h-vetinari) [inline recipe/build.sh:33]: Added.

**Review by h-vetinari — COMMENTED** [review]
2990d668b589962152aee3b2a434dd1b63856a14 already anticipates the switching off of the benchmarks from #143; otherwise it's mostly just grouping `_ENABLE_` and `_INCLUDE_` options.

**Comment by h-vetinari** [comment]
@isuruf
This is ready, with the caveat that GCC 10 didn't work on linux-64 due to a failing test. I raised an issue upstream: https://github.com/llvm/llvm-project/issues/53980

Despite the diff, the only real change is https://github.com/conda-forge/llvmdev-feedstock/pull/145/commits/6c2d36d31417bd554403784db6e655e0dbe938aa (and I can rebase out the various reverts etc. if desired).

**Review by isuruf — APPROVED** [review]

**Pull request merged**